### ISSUE TITLE
[Query] Making the inference more robust 

### ISSUE THREAD

**Issue by saurabhvyas** [CorentinJ/Real-Time-Voice-Cloning#48]
Hello, thank you for your work, I have read your thesis briefly, I liked the style of writing, and also, code quality is good, its easy for researchers and hobbyists to play with it, I have experimented with a few voices, and I have observed the following behaviors 

 1. sometimes, there is a space, empty sound in spectrogram and final audio, like pauses which are introduced, what could be the reason for it ? Is it because of the vocoder quality ?

2. The audio volume changes with text, for example, for some texts, the final audio has lower volume, but for some text, it has significantly higher volume, how can I fix this ? Is this also related to vocoder ( wavenet ) ?

Also, since you mention in your thesis, that each system ( of 3 systems ) can be trained independently. with independent datasets, If say, I want to finetune on a specific dataset, or even, a single person, who is native english speaker, so should I only retrained the vocoder system, or tacotron also needs to be retrained ( or finetuned ) ?

**Comment by CorentinJ** [comment]
Thanks. 

The quality of the final product is not that good, mostly due to the time limits that I had. The pauses are introduced by the synthesizer, not the vocoder. You can convince yourself of that by using griffin lim as vocoder. 

The output volume is a bug that is worth investigating. Essentially, the vocoder produces a few samples that are above 1 in absolute value so it screws with the normalization. Even when clamping all samples to [-1, 1], the issue persists.

If you want to finetune to single speaker, you should use another repo such as fatchord's or rayahane's, because this one is about voice cloning. For training on a specific dataset, you will need to retrain both the synthesizer and the vocoder. If in another language, the speaker encoder will need to be retrained as well.

**Comment by primejava** [comment]
in this link  https://github.com/NVIDIA/tacotron2/issues/162 ,Tugstugi said
he could fine-tune a Mongolian TTS using pre-trained English models.,I
think this is a very interesting thing.,But I'm a beginner. I don't know
how to start.Can this give you some inspiration?

Corentin Jemine <notifications@github.com> 于2019年7月16日周二 上午12:37写道：

**Comment by saurabhvyas** [comment]
Thank you, for your response, I will close this now, but anyone can continue related discussions if they want to
